### PR TITLE
Change Default Time Zone Handling

### PR DESCRIPTION
**Pull request by sowu880**
1. Change default time zone handling for FormatAsDateTime() API from "local" to "preserve".
2. Output of dateTime filters should be valid FHIR format with time zone.

### CODE REVIEW

**Review by qiwjin — APPROVED** [review]

**Review by BoyaWu10 — APPROVED** [review]

**Review by yankunhuang-pku — APPROVED** [review]

**Comment by yankunhuang-pku** [comment]
Do we need to update the unit test as well? It seems that the pipeline in Mac fails after this change.
`Failed Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.FunctionalTests.GivenHl7v2Message_WhenConverting_ExpectedFhirResourceShouldBeReturned(rootTemplate: "OML_O21", inputFile: "../../data/SampleData/Hl7v2/OML-O21-03.hl7", expectedFile: "TestData/Expected/Hl7v2/OML_O21/OML-O21-03-expected.json")
`

**Review by ACMoretxj — APPROVED** [review]

**Comment by yankunhuang-pku** [comment]
Why do we need to modify so many sample data and functional tests? It seems that in this [pipeline](https://microsofthealthoss.visualstudio.com/FhirConverter/_build/results?buildId=18538&view=results), all functional tests under windows can pass, and only one test in MAC fails

**Comment by sowu880** [comment]
> Why do we need to modify so many sample data and functional tests? It seems that in this [pipeline](https://microsofthealthoss.visualstudio.com/FhirConverter/_build/results?buildId=18538&view=results), all functional tests under windows can pass, and only one test in MAC fails

Since we changed the time zone handling methods( change 'local' to 'preserve'). Although all tests except one passed, the expected result is incorrect base our new datetime handling methods.

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/FunctionalTests.cs [hunk 1]
```diff
@@ -246,6 +246,9 @@ public void GivenCcdaDocument_WhenConverting_ExpectedFhirResourceShouldBeReturne
             var expectedContent = File.ReadAllText(expectedFile);
             var actualContent = ccdaProcessor.Convert(inputContent, rootTemplate, new TemplateProvider(templateDirectory, DataType.Ccda));
 
+            var newPath = expectedFile.Replace("Ccda", "newCcda");
+            Directory.CreateDirectory(Path.GetDirectoryName(newPath));
+            File.WriteAllText(newPath, actualContent);
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/FunctionalTests.cs:251]: Remove?

**Pull request merged**